elastic/datadog_checks/elastic/elastic.py

```diff
@@ -439,6 +461,123 @@ def _process_cat_allocation_data(self, admin_forwarder, version, base_tags):
+            else:
+                self.rate(metric_name, branch_value, tags=tags_to_submit, hostname=hostname)
+        else:
+            self.log.debug("Metric not found: %s -> %s", path, metric_name)
+
+    def _run_custom_queries(self, admin_forwarder, base_tags):
+        self.log.debug("Running custom queries")
+        custom_queries = self._config.custom_queries
+
+        for endpoints in custom_queries:
+            columns = endpoints.get('columns', [])
+            data_path = endpoints.get('data_path')
+            endpoint = endpoints.get('endpoint')
+            static_tags = endpoints.get('tags', [])
+            endpoint = self._join_url(endpoint, admin_forwarder)
+            payload = endpoints.get('payload', {})
+
+            data = self._get_data(endpoint, data=payload)
+
+            # If there are tags, add the tag path to list of paths to evaluate while processing metric
+            dynamic_tags = get_dynamic_tags(columns)
+            tags = base_tags + static_tags
+
+            for column in columns:
+                metric_type = column.get('type', 'gauge')
+
+                # Skip tags since already processed
+                if metric_type == 'tag':
+                    continue
+                name = column.get('name')
+                value_path = column.get('value_path')
+                if name and value_path:
+                    value = data
+
+                    # Traverse the nested dictionaries to the data_path
+                    for key in re.split(REGEX, data_path):
+                        if value is not None:
+                            value = value.get(key.replace('\\', ''))
+                        else:
+                            break
+
+                    # At this point, there may be multiple branches of value_paths.
+                    # If value is a list, go through each entry
+                    if isinstance(value, list):
+                        for branch in value:
+                            self._process_custom_metric(
+                                value=branch,
+                                data_path=data_path,
+                                value_path=value_path,
+                                dynamic_tags=dynamic_tags,
+                                xtype=metric_type,
+                                metric_name=name,
+                                tags=tags,
+                                hostname=None,
+                            )
+                    else:
+                        self._process_custom_metric(
+                            value=value,
+                            data_path=data_path,
+                            value_path=value_path,
+                            dynamic_tags=dynamic_tags,
+                            xtype=metric_type,
+                            metric_name=name,
+                            tags=tags,
+                            hostname=None,
+                        )
```

Comment:
I think we can avoid duplicating `self._process_custom_metric` by doing something like:

```

if isinstance(value, list):
  values = value
else:
  values = [value]

for cur_value in values:
  self._process_custom_metric(...)
```